### PR TITLE
Add ProPhoto RGB

### PR DESCRIPTION
**Pull request by tomcur**
For completeness, as in #35 with chromatic adaptation as in #36, the exact rational conversional matrices are

To linear sRGB

```
[    194437324536840809654921264985197050522635113 / 95576300929597858414530290868838637470430160         -334349093607417013044376331191497635943023 / 459501446776912780839087936869416526300145           -15292859747991764489911683478626450065757 / 49857225315387510910031450635805236030480]
[-1213686717324350640401526626143487705860808197 / 5303953588023484083939969999901748105291028600      62818873514175348299775154095551474067513974 / 50999553730995039268653557691362962550875275          -8097308360146843419467605577526078179247 / 2766798950455651582649958268076029267235800]
[     -916444166400653666849659340340134515395073 / 107080945084182313940532920349400661387210532        -315617386253206690895676451127897806279268 / 2059248943926582960394863852873089642061741              2237862560677598178438971725727906064237 / 1926158780497226520254940736232990869124]
```

From linear sRGB

```
[230208194708934909047266103904422197754053102 / 434945620713794670356359523418993609621703125  86159157180562584836685822742450446226228523 / 260967372428276802213815714051396165773021875  26202356015966622820550163547351714924543922 / 186405266020197715867011224322425832695015625]
[     733508889578727735346306717377661182359467 / 7456918390093613987566404091695576518625000     7816019310536291178779472833443402305449941 / 8948302068112336785079684910034691822350000    1260360450407861619423220078690480490343493 / 44741510340561683925398424550173459111750000]
[              2416241602970375872883355719587686988 / 143181794917587072365282753714076437025               10108011673285988197940323841248304012 / 85909076950552243419169652228445862215                 1831313308263153440825106279690762217 / 2115987117008675946284966803656302025]
```

### CODE REVIEW

File: color/src/colorspace.rs [hunk 1]
```diff
@@ -355,6 +355,69 @@ impl ColorSpace for A98Rgb {
     }
 }
 
+/// 🌌 The Prophoto RGB color space.
```
Comment (tomcur) [inline color/src/colorspace.rs:358]: Should probably be ProPhoto, but clippy complains: 

```
warning: item in documentation is missing backticks
   --> color/src/colorspace.rs:358:11
    |
358 | /// 🌌 The ProPhoto RGB color space.
    |            ^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
    = note: requested on the command line with `-W clippy::doc-markdown`
help: try
    |
358 | /// 🌌 The `ProPhoto` RGB color space.
    |            ~~~~~~~~~~
```

Add it to [valid idents](https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown), or slap an `#[expect(clippy::doc_markdown)]` on it?

File: color/src/colorspace.rs [hunk 2]
```diff
@@ -355,6 +355,69 @@ impl ColorSpace for A98Rgb {
...
+///
+///
+/// Prophoto RGB is similar to [sRGB](`Srgb`) but has higher red, green and blue chromaticities,
+/// thereby extending its gamut over sRGB on all components.
```
Comment (tomcur) [inline color/src/colorspace.rs:362]: Mention it has a different white point? We don't mention the white point of other color spaces, but as (linear) sRGB is central, maybe we should mention it here.

Comment (raphlinus) [reply color/src/colorspace.rs:362]: Yes, since it has a white point that differs from sRGB, it's worth stating, but we don't need to make a big deal about it.

File: color/src/colorspace.rs (hunk 1 above)
Comment (raphlinus) [reply color/src/colorspace.rs:358]: I believe our practice is valid idents, see for example [.clippy.toml](https://github.com/linebender/vello/blob/main/.clippy.toml) from Vello. And we should use the correct capitalization.

**Review by raphlinus — APPROVED** [review]
Looks good modulo the white point reference.

For information, I spot-checked the color pairs in the test against color.js. We probably want to get more systematic about this at some point, including testing the breakpoint between linear and nonlinear segments, and negative values for out-of-gamut colors.

**Comment by tomcur** [comment]
> For information, I spot-checked the color pairs in the test against color.js. We probably want to get more systematic about this at some point, including testing the breakpoint between linear and nonlinear segments, and negative values for out-of-gamut colors.

Absolutely.
